src/WorkplaceChannel.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace NotificationChannels\Workplace;
+
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\ClientException;
+use Illuminate\Notifications\Notification;
+use NotificationChannels\Workplace\Exceptions\CouldNotSendNotification;
+
+class WorkplaceChannel
+{
+    /** @var ClientInterface Http client. */
+    protected $httpClient;
+
+    public function __construct(ClientInterface $httpClient)
+    {
+        $this->httpClient = $httpClient;
+    }
+
+    /**
+     * Send the given notification.
+     *
+     * @param mixed $notifiable
+     * @param \Illuminate\Notifications\Notification $notification
+     *
+     * @throws \NotificationChannels\Workplace\Exceptions\CouldNotSendNotification
+     */
+    public function send($notifiable, Notification $notification)
+    {
+        if (! $to = $notifiable->routeNotificationFor('workplace', $notification)) {
+            throw CouldNotSendNotification::endpointNotProvided();
+        }
+
+        $message = $notification->toWorkplace($notifiable);
```

Comment:
Is it worth checking that this method exists?
Also, something a few other channels do is if $message is a string, they just make a new plain message out of it (so the notification can just return a string and the channel will convert to a message)

Reply:
I don't understand what do you mean by checking if the method exists. I've checked 5 or 6 other packages and no one seems to be checking for any method existence, but maybe I misunderstood you.
The `WorkplaceChannel` can now work with both `string/WorkplaceMessage` coming from `toWorkplace()` method. I've also added a test for this.